query-engine/core/src/schema_builder/input_type_builder/update_input_type_extension.rs

```diff
@@ -112,11 +104,14 @@ pub trait UpdateInputTypeBuilderExtension<'a>: InputTypeBuilderBase<'a> + Create
                         }
                     };
 
-                    Some(input_field(
-                        rf.name.clone(),
-                        InputType::opt(InputType::null(InputType::object(input_object))),
-                        None,
-                    ))
+                    let field_type = InputType::object(input_object);
+                    let field_type = if rf.is_optional() {
+                        InputType::opt(InputType::null(field_type))
```

Comment:
Question: does it make sense to allow `null` when the relation is optional? The intent would be to disconnect the relation. Does this work too, or is this always supposed be done through `disconnect`?

Reply:
`null` inputs for relations are allowed but ignored. Not allowing nulls now would be a breaking change, I'd rather not do that. I do agree though that disallowing it is the better choice (user expectation would be that something happens if you provide an input), but that's for product to decide.

Reply:
:+1: I'll coordinate with Jan (Hervé and Alberto are on vacation)

Reply:
> does it make sense to allow null when the relation is optional? 

@tomhoule are we talking about allowing null in the virtual relation field (e.g. `User`) or the foreign key (e.g. `userId`)? I'd love a Prisma JS Client example of the two cases if possible.

> user expectation would be that something happens if you provide an input

@dpetrick Can you clarify what you mean by this?

Reply:
@matthewmueller this is not something general, only in the update case. Given the following schema:

```prisma
model Dog {
  id Int @id
  favouriteToyId Int
  favouriteToy Toy @relation(fields: [favouriteToyId], references: [id])
}

model Toy {
  id Int @id
  name String
}
```

This query is currently accepted and does not update anything, but it would be rejected after this PR (pseudo-code):

```typescript
prisma.dog.update({
  where: {id: 10},
  data: {
    favouriteToy: null
  },
})
```

Reply:
It happened multiple times that users provided `null` for a relation in an update in the expectation that this would disconnect an optional relation, which is not the case. We require calls to `disconnect`, not `null`.

```graphql
mutation {
  updateModelX(where: ..., data: { relatedOptModel: null }) {
    ...
  }
}
```

Reply:
Thanks for clarifying, so rejecting `null` would indeed be a breaking change, but it wasn't working as expected anyways? It was just ignored.

Do I have that right?

Reply:
That's my understanding, yes.